File: src/test/java/com/dsobko/credible/test/CucumberStep.java
package com.dsobko.credible.test;

import org.springframework.test.annotation.DirtiesContext;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.boot.test.context.SpringBootTest;

import java.lang.annotation.*;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
@SpringBootTest
@ContextConfiguration(classes = MainConfiguration.class)
@DirtiesContext
public @interface CucumberStep {
}

File: src/test/java/com/dsobko/credible/test/CucumberStep.kt
package com.dsobko.credible.test

import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.DirtiesContext
import org.springframework.test.context.ContextConfiguration
import java.lang.annotation.Inherited

@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
@SpringBootTest
@ContextConfiguration(classes = arrayOf(MainConfiguration::class))
@DirtiesContext
annotation class CucumberStep
